File: Anul II/Medii de proiectare si programare/Laborator/lab03/C#/Ubb.BikeContest.Repository/src/IRepository.cs
using System.Collections;
using System.Collections.ObjectModel;

namespace Ubb.BikeContest.Repository;

public interface IRepository<TId, TEntity> where TEntity : Ubb.BikeContest.Model.Identifiable<TId>
{
    TEntity Get(TId id);

    IEnumerable Read();

    void Add(TEntity newEntity);

    void Delete(TId id);

    void Update(TEntity updatedEntity);
}

File: Anul II/Medii de proiectare si programare/Laborator/lab09/persistence/src/main/java/com/ubb/repository/IRepository.java
package com.ubb.repository;

import com.ubb.model.Identifiable;

import java.util.Collection;
import java.util.Optional;

public interface IRepository<ID, T extends Identifiable<ID>> {
    Optional<T> findById(final ID id);

    Collection<T> findAll();

    void save(final T newEntity);

    void delete(final ID id);

    void update(final T updatedEntity);
}
